Reduce a bit of code repetition around directory creation

alexwlchan opened the pull request.

bcail commented:
I see you restored the _create_cache_dir method. It looks to me like it could just be removed, along with its test. Is there a reason to still keep it around?

alexwlchan commented:
> Is there a reason to still keep it around?

Yeah, now I look again, I can’t think of one. (I hadn’t noticed it was unused after my patch.) I’ll remove it.

alexwlchan committed:
Consolidate code for creating directories into a utils method

Also remove the _create_cache_dir method, which at this point can just
be replaced by a call straight to the utility.

alexwlchan committed:
Remove one more use of standalone makedirs
